aptoide-installer/src/main/java/cm/aptoide/pt/installer/AptoideInstallPackageInfoMapper.kt

```diff
@@ -2,67 +2,106 @@ package cm.aptoide.pt.installer
 
 import android.os.Environment
 import cm.aptoide.pt.feature_apps.data.App
+import cm.aptoide.pt.feature_apps.data.File
 import cm.aptoide.pt.feature_apps.domain.AppMetaUseCase
+import cm.aptoide.pt.feature_apps.domain.DynamicSplitsUseCase
 import cm.aptoide.pt.install_info_mapper.domain.InstallPackageInfoMapper
 import cm.aptoide.pt.install_manager.dto.InstallPackageInfo
 import cm.aptoide.pt.install_manager.dto.InstallationFile
-import cm.aptoide.pt.install_manager.dto.InstallationFile.Type.OBB_PATCH
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class AptoideInstallPackageInfoMapper @Inject constructor(
   private val appMetaUseCase: AppMetaUseCase,
+  private val dynamicSplitsUseCase: DynamicSplitsUseCase,
 ) : InstallPackageInfoMapper {
   override suspend fun map(app: App): InstallPackageInfo = InstallPackageInfo(
     versionCode = app.versionCode.toLong(),
     installationFiles = mutableSetOf<InstallationFile>()
       .apply {
-        with(
-          app.file.takeIf { it.path.isNotEmpty() }
-            ?: appMetaUseCase.getMetaInfo(app.packageName).file
-        ) {
-          add(
-            InstallationFile(
-              name = fileName,
-              type = InstallationFile.Type.BASE,
-              md5 = md5,
-              fileSize = filesize,
-              url = path,
-              altUrl = path_alt,
-              localPath = Environment.getExternalStorageDirectory().absolutePath + "/.aptoide/"
-            )
-          )
+        val appMeta = appMetaUseCase.getMetaInfo(app.packageName)
+
+        add(appMeta.file.toInstallationFile(InstallationFile.Type.BASE))
+
+        appMeta.obb?.run {
+          add(main.toInstallationFile(InstallationFile.Type.OBB_MAIN))
+          patch?.let { add(it.toInstallationFile(InstallationFile.Type.OBB_PATCH)) }
         }
-        app.obb?.run {
-          with(main) {
-            add(
-              InstallationFile(
-                name = fileName,
-                type = InstallationFile.Type.OBB_MAIN,
-                md5 = md5,
-                fileSize = filesize,
-                url = path,
-                altUrl = path_alt,
-                localPath = Environment.getExternalStorageDirectory().absolutePath + "/.aptoide/"
-              )
-            )
+
+        appMeta.aab?.run {
+          requiredSplitTypes.forEach { type ->
+            if (splits.find { it.type == type } == null) {
+              throw IllegalStateException("AAB required split types not found")
```

Comment:
I don't get this check and this message, what does it mean? 🤔

Reply:
It is to detect possible errors in the API response, where the list of splits doesn't contain the splits required. For example, it might require an ABI split ask, but the list of splits doesn't actually contain it which would be an error

Reply:
Ok, I see now.
Could you, please, make this message reflect this case, so an "alien" would understand what exactly is wrong here? 
Next year we would be those "aliens" to ourselves, or any newcomer in our team :-)

Reply:
Added a comment explaining this